### PR TITLE
+ fetch current pod ip from socket mark

### PR DESCRIPTION
**Pull request by kebe7jun**
https://github.com/merbridge/merbridge/issues/117  part 1.

### CODE REVIEW

**Comment by kebe7jun** [comment]
@dddddai PTAL.

File: bpf/mb_connect.c [hunk 1]
```diff
@@ -115,28 +154,37 @@ static inline int tcp4_connect(struct bpf_sock_addr *ctx)
...
         };
-        void *curr_ip = bpf_map_lookup_elem(&process_ip, &pid);
-        if (curr_ip) {
-            // envoy to other envoy
-            if (*(__u32 *)curr_ip != ctx->user_ip4) {
-                debugf("enovy to other, rewrite dst port from %d to %d",
-                       ctx->user_port, IN_REDIRECT_PORT);
+        if (curr_pod_ip) {
+            origin.flags |= 1;
+            if (curr_pod_ip == ip) {
```
Comment (dddddai) [inline bpf/mb_connect.c:159]: Shouldn't it be `curr_pod_ip != ip`?

File: bpf/mb_connect.c [hunk 2]
```diff
@@ -67,6 +67,30 @@ static inline int tcp4_connect(struct bpf_sock_addr *ctx)
...
+            if (ip) {
+                curr_pod_ip = *ip; // network order
+            } else {
+                debugf("get ip for mark %x error", curr_ip_mark);
+            }
+        }
+    }
+    if (curr_pod_ip == 0) {
+        debugf("get current pod ip error");
+        return 1;
```
Comment (dddddai) [inline bpf/mb_connect.c:92]: So for the pods running before merbridge, the dst port will never get rewrited?

We have 2 solutions to solve this:
1. use the legacy logic, then we should not return at this point
2. listen on `SOCK_IP_MARK_PORT` in all injected pods when merbridge starts, then I think we can remove the legacy logic

Comment (kebe7jun) [reply bpf/mb_connect.c:92]: error logic. hhh. copy code error.

**Review by dddddai — APPROVED** [review]

**Pull request merged**